presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1509,6 +1510,15 @@ private static OGCGeometry geometryFromText(Slice input)
         catch (IllegalArgumentException e) {
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid WKT: " + input.toStringUtf8(), e);
         }
+
+        try {
+            // Test whether geometry is valid (can be serialized)
+            geometry.asBinary();
```

Comment:
This is likely going to be a big performance hit.

Reply:
https://github.com/prestosql/presto/issues/1138#issuecomment-512580455

Reply:
Problem is that ST_GeometryFromText() accepts invalid geometry right now (for example polygon with 2 coordinates). When ```GeometryType``` is returned in query exception is thrown (because ```OGCGeometry.asText()``` could not represent invalid geometry as WKT.

I've checked against postgis and it disallows creating invalid geometries at all with ```ST_GeometryFromText``` function.

Reply:
FWIW, the WKT syntax allows for polygons with one or two points. 

The JTS library validates that the geometries are valid upon parsing. We should consider switching to that library for parsing geometries.

Reply:
I agree with syntax but still that makes this geometry invalid (and it's now validated upon parsing right now).